### PR TITLE
table: adjust table id calculation.

### PR DESCRIPTION
**Pull request by disksing**
In order to allow tikv to support region merge easily, now only split at the begining of a region. 

ref https://github.com/pingcap/tidb/pull/4926

### CODE REVIEW

File: table/namespace_classifier.go [hunk 1]
```diff
@@ -144,7 +144,7 @@ func (c *tableNamespaceClassifier) GetRegionNamespace(regionInfo *core.RegionInf
 	c.RLock()
 	defer c.RUnlock()
 
-	tableID := c.getTableID(regionInfo)
+	tableID := c.tableIDDecoder.DecodeTableID(regionInfo.StartKey)
```
Comment (siddontang) [inline table/namespace_classifier.go]: @disksing 

Do we need to set metadata region to the special namespace later?

Comment (disksing) [reply table/namespace_classifier.go]: Good idea.

Comment (siddontang) [reply table/namespace_classifier.go]: File an issue? @disksing

**Comment by siddontang** [comment]
LGTM

**Pull request merged**